packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -143,58 +143,44 @@ export const AuthContextProvider = ({
     setLoginState({ trigger: AuthTriggers.LegacyLogout });
   }
 
-  const authContext: AuthContextData = useMemo(
-    () => ({
-      isAuthReady: !isNullOrUndefined(firstLoad),
-      user: endUser,
-      isLoggedIn: !!endUser?.id,
-      referral: loginState?.referral ?? referral,
-      referralOrigin: loginState?.referralOrigin ?? referralOrigin,
-      firstVisit: user?.firstVisit,
-      trackingId: user?.id,
-      shouldShowLogin: loginState !== null,
-      showLogin: ({ trigger, options = {} }) => {
-        const hasCompanion = !!isCompanionActivated();
-        if (hasCompanion) {
-          const signup = `${process.env.NEXT_PUBLIC_WEBAPP_URL}signup?close=true`;
-          window.open(signup);
-        } else {
-          setLoginState({ ...options, trigger });
-        }
-      },
-      closeLogin: () => setLoginState(null),
-      loginState,
-      updateUser,
-      logout,
-      loadingUser,
-      tokenRefreshed,
-      loadedUserFromCache,
-      getRedirectUri,
-      anonymous: user,
-      visit,
-      isFetched,
-      refetchBoot,
-      deleteAccount,
-      accessToken,
-      squads,
-    }),
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      user,
-      loginState,
-      isFetched,
-      loadingUser,
-      tokenRefreshed,
-      loadedUserFromCache,
-      visit,
-      accessToken,
-      squads,
-      firstLoad,
-    ],
-  );
-
   return (
-    <AuthContext.Provider value={authContext}>{children}</AuthContext.Provider>
+    <AuthContext.Provider
+      value={{
+        isAuthReady: !isNullOrUndefined(firstLoad),
+        user: endUser,
+        isLoggedIn: !!endUser?.id,
+        referral: loginState?.referral ?? referral,
+        referralOrigin: loginState?.referralOrigin ?? referralOrigin,
+        firstVisit: user?.firstVisit,
+        trackingId: user?.id,
+        shouldShowLogin: loginState !== null,
+        showLogin: useCallback(({ trigger, options = {} }) => {
+          const hasCompanion = !!isCompanionActivated();
+          if (hasCompanion) {
+            const signup = `${process.env.NEXT_PUBLIC_WEBAPP_URL}signup?close=true`;
+            window.open(signup);
+          } else {
+            setLoginState({ ...options, trigger });
+          }
+        }, []),
+        closeLogin: useCallback(() => setLoginState(null), []),
+        loginState,
+        updateUser,
+        logout,
+        loadingUser,
+        tokenRefreshed,
+        loadedUserFromCache,
+        getRedirectUri,
+        anonymous: user,
+        visit,
+        isFetched,
+        refetchBoot,
+        deleteAccount,
+        accessToken,
+        squads,
+      }}
```

Comment:
If changes were unnecessary why do it? Couldn't we have removed the lint disable by having this in a variable still?

Reply:
The useMemo isn't really useful to put everything in it. We prefer using memoization only on places that is necessary.

Reply:
Yeah not about memoization, just about this being in a separate variable

Reply:
Right. That part, I just thought we wouldn't use it anywhere from here, so I just applied it directly. I don't have any preference though, so either way works for me.

Reply:
Merging now to unblock so we can immediately build an extension as this is pretty important.

If we ever decide otherwise, I can open another PR.